code/FsComposite/Composite.Core/Processing.fs

```diff
@@ -4,22 +4,51 @@ open Composite.Core.Composite
 
 module Processing =
     
+    let rec applyScnToElement scn acc element =
+        match scn with
+        | [] -> acc
+        | f :: scn_tail -> match f element with
+                           | Cons(Nil, Nil) -> applyScnToElement scn_tail acc element
+                           | x -> applyScnToElement scn_tail (LazyList.append acc x) element
+
     let rec fold_2 acc f_update_acc lst =
         match acc with
         |(None, _) | (_, None) -> match lst with
                                     | Nil -> acc
                                     | Cons(h, tail) -> fold_2 (f_update_acc h acc) f_update_acc tail
         | _ -> acc
 
+    let rec fold_3 acc f_update_acc lst =
+        match acc with
+        | (_, None, None) | (None, _, None) | (None, None, _) 
+        | (None, _, _) | (_, None, _) | (_, _, None) -> match lst with
+                                                        | Nil -> acc
+                                                        | Cons(h, tail) -> fold_3 (f_update_acc h acc) f_update_acc tail
+        | _ -> acc

```

Comment:
A version of these should be devised to support the new list-oriented one-element-at-a-time style of `cata`.